tests/experimental/test_lightning_qubit_2.py

```diff
@@ -0,0 +1,331 @@
+
+    def test_tracker_set_upon_initialization(self):
+        """Test that a new tracker is initialized with each device."""
+        assert LightningQubit2.tracker is not LightningQubit2().tracker
+
+    def test_tracker_not_updated_if_not_active(self):
+        """Test that the tracker is not updated if not active."""
+        dev = LightningQubit2()
+        assert len(dev.tracker.totals) == 0
+
+        dev.execute(qml.tape.QuantumScript())
+        assert len(dev.tracker.totals) == 0
+        assert len(dev.tracker.history) == 0
+
+    def test_tracking_batch(self):
+        """Test that the experimental qubit integrates with the tracker."""
+
+        qs = qml.tape.QuantumScript([], [qml.expval(qml.PauliZ(0))])
+
+        dev = LightningQubit2()
+        with qml.Tracker(dev) as tracker:
+            dev.execute(qs)
+            dev.execute([qs, qs])  # and a second time
+
+        assert tracker.history == {"batches": [1, 1], "executions": [1, 2]}
+        assert tracker.totals == {"batches": 2, "executions": 3}
+        assert tracker.latest == {"batches": 1, "executions": 2}
+
+
+@pytest.mark.skipif(not CPP_BINARY_AVAILABLE, reason="Lightning binary required")
+class TestSupportsDerivatives:
+    """Test that LightningQubit2 states what kind of derivatives it supports."""
+
+    def test_supports_adjoint(self):
+        """Test that LightningQubit2 says that it supports adjoint method."""
+        dev = LightningQubit2()
+        config = ExecutionConfig(gradient_method="adjoint")
+        assert dev.supports_derivatives(config) is True
+
+        qs = qml.tape.QuantumScript([], [qml.state()])
+        assert dev.supports_derivatives(config, qs)
+
+    @pytest.mark.parametrize(
+        "gradient_method", ["backprop", "parameter-shift", "finite-diff", "device"]
+    )
+    def test_does_not_support_other_gradient_methods(self, gradient_method):
+        """Test that LightningQubit2 currently does not support other gradient methods natively."""
+        dev = LightningQubit2()
+        config = ExecutionConfig(gradient_method=gradient_method)
+        assert not dev.supports_derivatives(config)
+
+
+@pytest.mark.skipif(not CPP_BINARY_AVAILABLE, reason="Lightning binary required")
+class TestBasicCircuit:
+    """Tests a basic circuit with one RX gate and two simple expectation values."""
+
+    def test_basic_circuit_numpy(self):
+        """Test execution with a basic circuit."""
+        phi = np.array(0.397)
+        qs = qml.tape.QuantumScript(
+            [qml.RX(phi, wires=0)], [qml.expval(qml.PauliY(0)), qml.expval(qml.PauliZ(0))]
+        )
+
+        dev = LightningQubit2()
+        result = dev.execute(qs)
+
+        assert isinstance(result, tuple)
+        assert len(result) == 2
+
+        assert np.allclose(result[0], -np.sin(phi))
+        assert np.allclose(result[1], np.cos(phi))
+
+
+@pytest.mark.xfail  # [Return before Merge]: xfail for now. Waiting for core update.
```

Comment:
This test is expected to fail until [Issue 3967](https://github.com/PennyLaneAI/pennylane/issues/3967) is solved.

Reply:
The problem found here is being tackled in PennyLane [PR 3985](https://github.com/PennyLaneAI/pennylane/pull/3985).

Reply:
Thank you, @timmysilv!